lib/backend/dynamo/dynamodbbk.go

```diff
@@ -572,6 +578,10 @@ func (b *Backend) Delete(ctx context.Context, key []byte) error {
 // ConditionalUpdate updates the matching item in Dynamo if the provided revision matches
 // the revision of the item in Dynamo.
 func (b *Backend) ConditionalUpdate(ctx context.Context, item backend.Item) (*backend.Lease, error) {
+	if item.Revision == backend.BlankRevision {
+		item.Revision = ""
+	}
+
```

Comment:
If the passed-in revision is the empty string we should fail the operation (I don't know if it should be a `CompareFailed` or a `BadParameter`, but `pgbk` will return `CompareFailed` on anything that's not a UUID, including the blank string).